crates/hir-def/src/body/lower.rs

```diff
@@ -437,9 +446,85 @@ impl ExprCollector<'_> {
                 let expr = self.collect_expr_opt(e.expr());
                 self.alloc_expr(Expr::Await { expr }, syntax_ptr)
             }
-            ast::Expr::TryExpr(e) => {
-                let expr = self.collect_expr_opt(e.expr());
-                self.alloc_expr(Expr::Try { expr }, syntax_ptr)
+            ast::Expr::TryExpr(e) => 'try_result: {
```

Comment:
nit but let's split this arm out into its own function given it became so big now